packages/core/src/utils/applyScopeDataToEvent.ts

```diff
@@ -113,24 +112,20 @@ export function mergeArray<Prop extends 'breadcrumbs' | 'fingerprint'>(
 function applyDataToEvent(event: Event, data: ScopeData): void {
   const { extra, tags, user, contexts, level, transactionName } = data;
 
-  const cleanedExtra = dropUndefinedKeys(extra);
-  if (Object.keys(cleanedExtra).length) {
-    event.extra = { ...cleanedExtra, ...event.extra };
+  if (Object.keys(extra).length) {
```

Comment:
I don't _think_ we really have to do all of this here 🤔

Reply:
I mean, `JSON.stringify()` will eventually def. drop this 🤔 Let's see if we have tests covering this/if we need to add this!

Reply:
how would you test this? 🤔 all our tests pass, and I think from an event perspective, it should not matter because it will never be serialized:

```
JSON.stringify({ blub: undefined })
--> '{}'
```

So I am not quite sure how we could test this nicely/better than we already do? 🤔

Reply:
Yeah you're right, I think we are fine.